packages/devextreme/js/__internal/ui/stepper/stepper.ts

```diff
@@ -109,12 +112,26 @@ class Stepper extends CollectionWidgetAsync<StepperProperties> {
 
     $indicatorElement.prependTo($container);
 
-    if (isDefined(title)) {
-      const $stepTitleDiv = $('<div>').addClass(STEP_TITLE_CLASS);
+    if (isDefined(title) || optional) {
+      const $stepLabelDiv = $('<div>').addClass(STEP_LABEL_CLASS);
 
-      $stepTitleDiv.text(title);
+      if (isDefined(title)) {
+        const $stepTitleDiv = $('<div>').addClass(STEP_TITLE_CLASS);
 
-      $stepTitleDiv.appendTo($container);
+        $stepTitleDiv.text(title);
+
+        $stepTitleDiv.appendTo($stepLabelDiv);
```

Comment:
Let's think how we could simplify render function. Now it looks overcomplecated, I believe we can refactor it